### PR TITLE
Move asfortranarray to core

### PR DESCRIPTION
**Pull request by yuyu2172**
This PR moves`cupy.asfortranarray` under cupy.core` just like `cupy.ascontiguousarray`.
Also, the function now detects the case when copy is not necessary. `cupy.ascontiguousarray` does this conditional handling too.

### CODE REVIEW

File: cupy/core/core.pyx [hunk 1]
```diff
@@ -1742,6 +1742,45 @@ cpdef ndarray ascontiguousarray(ndarray a, dtype=None):
...
+        dtype = a.dtype
+    else:
+        dtype = numpy.dtype(dtype)
+        if a._f_contiguous and dtype == a.dtype:
+            return a
+
+    newarray = ndarray(a.shape, dtype, order='F')
+    if (a.flags.c_contiguous and
+            (a.dtype == numpy.float32 or a.dtype == numpy.float64) and
+            a.ndim == 2 and dtype is None):
```
Comment (gwtnb) [inline cupy/core/core.pyx]: Always `dtype is not None` https://github.com/pfnet/chainer/pull/2176/files#diff-bebe923819e806ae22e9923de58e954cR1753

**Comment by gwtnb** [comment]
Thank you for your PR. Could you read the above review?

**Comment by yuyu2172** [comment]
Thank you. I fixed the code.

File: cupy/core/core.pyx [hunk 2]
```diff
@@ -1742,6 +1742,45 @@ cpdef ndarray ascontiguousarray(ndarray a, dtype=None):
...
+        dtype = a.dtype
+    else:
+        dtype = numpy.dtype(dtype)
+        if a._f_contiguous and dtype == a.dtype:
+            return a
+
+    newarray = ndarray(a.shape, dtype, order='F')
+    if (a.flags.c_contiguous and
+            (a.dtype == numpy.float32 or a.dtype == numpy.float64) and
+            a.ndim == 2):
```
Comment (gwtnb) [inline cupy/core/core.pyx]: `dtype == a.dtype` should be checked.

Comment (yuyu2172) [inline cupy/core/core.pyx]: My bad. I will fix it.

**Comment by gwtnb** [comment]
Thank you!

**Pull request merged**